packages/core/test/unit/query-generator/attributes-to-sql.test.ts

```diff
@@ -0,0 +1,221 @@
+      default: { id: 'INTEGER COMMENT Test' },
+      'mariadb mysql snowflake': { id: `INTEGER COMMENT 'Test'` },
+      mssql: { id: 'INTEGER NULL COMMENT Test' },
+      'sqlite ibmi': { id: 'INTEGER' },
+    });
+  });
+
+  it(`{ id: { type: 'INTEGER', unique: true } }`, () => {
+    expectPerDialect(() => queryGenerator.attributesToSQL({ id: { type: 'INTEGER', unique: true } }), {
+      default: { id: 'INTEGER UNIQUE' },
+      mssql: { id: 'INTEGER NULL UNIQUE' },
+    });
+  });
+
+  it(`{ id: { type: 'INTEGER', unique: true, comment: 'This is my comment' } }`, () => {
+    expectPerDialect(() => queryGenerator.attributesToSQL({ id: { type: 'INTEGER', unique: true, comment: 'This is my comment' } }), {
+      default: { id: 'INTEGER UNIQUE COMMENT This is my comment' },
+      'mariadb mysql snowflake': { id: `INTEGER UNIQUE COMMENT 'This is my comment'` },
+      mssql: { id: 'INTEGER NULL UNIQUE COMMENT This is my comment' },
+      'sqlite ibmi': { id: 'INTEGER UNIQUE' },
+    });
+  });
+
+  it(`{ id: { type: 'INTEGER', unique: true, comment: 'This is my comment' } }, { context: 'addColumn', key: 'column', table: { schema: 'foo', tableName: 'bar' } }`, () => {
+    expectPerDialect(() => queryGenerator.attributesToSQL({ id: { type: 'INTEGER', unique: true, comment: 'This is my comment' } }, { context: 'addColumn', key: 'column', table: { schema: 'foo', tableName: 'bar' } }), {
+      default: { id: 'INTEGER UNIQUE COMMENT This is my comment' },
+      'mariadb mysql snowflake': { id: `INTEGER UNIQUE COMMENT 'This is my comment'` },
+      postgres: { id: `INTEGER UNIQUE; COMMENT ON COLUMN "foo"."bar"."column" IS 'This is my comment'` },
+      mssql: { id: 'INTEGER NULL UNIQUE COMMENT This is my comment' },
+      'sqlite ibmi': { id: 'INTEGER UNIQUE' },
+    });
+  });
+
+  it(`{ id: { type: 'INTEGER', after: 'Bar' } }`, () => {
+    // @ts-expect-error -- after is not accepted in AttributeOptions so attributesToSQL might need to be typed differently
+    expectPerDialect(() => queryGenerator.attributesToSQL({ id: { type: 'INTEGER', after: 'Bar' } }), {
+      default: { id: 'INTEGER' },
+      'mariadb mysql': { id: 'INTEGER AFTER `Bar`' },
+      mssql: { id: 'INTEGER NULL' },
+      'snowflake ibmi': { id: 'INTEGER AFTER "Bar"' },
+    });
+  });
+
+  // TODO: check what this test is supposed to do and update the possible defaultValue accordingly
+  it(`No empty array as default value for BLOB allowed for some dialects`, () => {
+    expectPerDialect(() => queryGenerator.attributesToSQL({ id: { type: 'BLOB', defaultValue: [] } }), {
+      default: { id: 'BLOB' },
+      'postgres sqlite mssql db2': new Error('Could not guess type of value [] because it is an empty array'),
+    });
+  });
+
+  // TODO: this test is broken for at least DB2. It adds the default value because the data type is a string, which can't have special behaviors.
+  //  change type to an actual Sequelize DataType & re-enable
```

Comment:
In this instance it's honestly `default` that's broken. String types don't give any information so the default value should always be added.

If `DataTypes.BLOB` were used, then the dialect could enforce specific behaviors